cache/transaction_registry_redis.go

```diff
@@ -13,14 +13,20 @@ import (
 
 type redisTransactionRegistry struct {
 	redisClient redis.UniversalClient
-	// deadline specifies TTL of record to be kept, no matter the associated transaction status
+
+	// deadline specifies TTL of the record to be kept
 	deadline time.Duration
+
+	// transactionEndedDeadline specifies TTL of the record to be kept that has been ended (either completed or failed)
+	transactionEndedDeadline time.Duration
 }
 
-func newRedisTransactionRegistry(redisClient redis.UniversalClient, deadline time.Duration) *redisTransactionRegistry {
+func newRedisTransactionRegistry(redisClient redis.UniversalClient, deadline time.Duration,
```

Comment:
becareful, since you're modifying this struct that is seriailzed into redis, you need to update the key version, otherwise people will have deserialisation issue after update

Reply:
I don't modify objects that are stored in redis. This is just a bookkeping structure implementing transaction registry interface

Reply:
you're right, my bad